### DISCUSSION TITLE
Cannot get "excluded-libraries" to work in config.yml

### DISCUSSION THREAD

**Discussion by ztevo** [Taxel/PlexTraktSync#1887] [Support] [answered]
Hi 

### Issue/question
I am trying to exclude a specific library in my config.yml file called "Specials (s1)".

However, when I run the Sync command, the "Specials (s1)" library is still being synced.  Please can you help?

### Background info

Except from Info command, showing the library names:
```
Plex Server version: 1.40.1.8227-c0dd5a73e, updated at: 2024-04-02 14:37:45
Enabled 3 libraries in Plex Server:
 - 2: TV (s1)
 - 3: Movies (s1)
 - 4: Specials (s1)
```

Excerpt from my config.yml file:
```
cache:
  path: $PTS_CACHE_DIR/trakt_cache

excluded-libraries:
  - 'Specials (s1)'

config:
  dotenv_override: true

plex:
  timeout: 30
```

Except when I run PlexTraktSync Sync:
```
INFO     PlexTraktSync [0.29.12]
INFO     Connecting with url: https://172-17-0-3.1ed892cc4ca4470b9368447c8537bd56.plex.direct:32400, timeout 30 seconds
INFO     Sync Movie sections: ['Movies (s1)', 'Specials (s1)']
INFO     Sync Show sections: ['TV (s1)']
Processing Movies (s1) 100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 112/112  [ 0:00:01 < 0:00:00 , 93 it/s ]
INFO     Movies (s1) processed in 1.5 seconds
WARNING  Buccaneers Copy (2023): Skipping <PlexGuid:com.plexapp.agents.none://c7f34ffc575a6c7d4e0ca8b7211c7b8f4997526e?lang=xn> because provider 'none' has no external Id

```
_It will then chuck up warnings for all items in Specials (s1), because they are a collection of videos without id's.  For example - "Buccaneers Copy (2023)" in the above excerpt._

### Troubleshooting steps I have tried:
- Various combinations of quotation marks (double quotes, single quotes etc)
- Various combinations of case sensitivity in the name - "Specials (s1)" is the correct case
- Self updating to latest  version
- I also ran the bug-report command (not submitted!). This did something interesting, as I found in the bug report the capitalisation and quotation marks used in excluded-libraries DOES NOT MATCH what is in the actual config.yml file - See below excerpt from bug report:

```
cache:
  path: /home/ubuntu/.cache/PlexTraktSync/trakt_cache
excluded-libraries:
- Specials (S1)
config:
  dotenv_override: true
plex:
  timeout: 30
```

NB - I understand I can edit the servers.yml file but would prefer not to if avoidable!

Thanks in advance
Steve

**Comment by glensc** [comment]
are you sure you're editing correct config file? do other changes have effect? also, using libraries config in servers.yml works?

I'm not aware why your s1 is transformed to S2, there should not be any code in place to do that.

**Reply by ztevo** [reply]
Thanks - 99% sure it is the correct config file, as I am copying the path from the "info" command. Also I have made modifications to the config file in the past which have applied successfully.

I will double-check tomorrow, by modifying other options in the config file and seeing if they apply successfully. If so, then it will confirm the config file location hasn't somehow been modified.

I believe servers.yml file works as I modified it in the past to test, but will try again tomorrow again to double-check and report back.

**Comment by ztevo** [comment]
Hi - sorry it has taken me a while to come back to this. I can confirm:

1. Modifying the servers.yml file does work as expected
2. Modifying any part of the config.yml file does NOT work expected

In regard to point 2 - I tried disabling rating sync, however the ratings still sync.  So it seems the issue not with the excluded-libraries settings specifically.

Please could you help advise any troubleshooting steps? (I have tried _pipx uninstall plextraktsync_ and reinstalling via _pipx install plextraktsync_ - this didn't fix it.)

- Could PTS be pulling a cached version of the config file perhaps?
- Or be calling on config.yml that is saved in a different location?


For reference, I am using the same config.yml listed in command _plextraktsync info_ (i.e.  /home/ubuntu/.config/PlexTraktSync/config.yml)

Many thanks!

**Comment by ztevo** [comment]
Just to add:

I just learned there is a _plextraktsync config_ command.  When I make changes to the config.yml file, the changes are NOT replicated in this command.

I have tried using _plextraktsync config --edit_ as alternative to editing the file with nano, but this does not seem to do anything...

Thanks again in advance

**Reply by glensc** [reply]
you have not gave info where you run it, terminal, x11, whatever? or what output did you got by running what command. anyway --edit and --locate use external commands (xdg-open on linux I guess?), so if those fail, then that's it.

i don't know what else to say. it should work.

make full reproducible details, then could have look, file paths, file contents, environment, etc.  maybe try it yourself first, create new unix user and setup under it documenting the steps. 

also, reinstalling pipx app should have no effect, it doesn't remove any configs or caches. also there's no cache for config.

**Reply by glensc** [reply]
for example, you haven't shared your servers.yml at all. 

for example, from readme:

> Additionally, you can list only libraries to be processed, in this case global `excluded-libraries` will not be used for this server.

so, what's in your servers.yml, does matter.

**Comment by ztevo** [comment, marked as answer]
Thank you for the guidance.

Well at some point during my troubleshooting, the _config.yml_ file was reset to default and then started working again (even with subsequent edits).

Closing - thanks again.

SOLUTION: Reset config.yml to default, then update parameters as required.
